Makefile

```diff
@@ -124,8 +127,21 @@ docs: typedoc generated-documentation
 view-docs:
 	PYTHONPATH=${ROOTDIR} PIPENV_IGNORE_VIRTUALENVS=1 expect -c 'spawn pipenv run mkdocs serve; expect "Serving on"; exec open "http://localhost:8000"; interact'
 
+###############################################################################
+### Deployment of documentation ### 
+
+# This step generates all the documentation for the SDK using mkdocs and dumps the contents in /site
+.PHONY: build-docs
```

Comment:
Might want to call this `build-mkdocs`. There's already a `docs` target that "builds" documentation, so we should probably just be specific about what each target actually does.

Alternatively, do we want to add this step to the `docs` target, or is unnecessary to do on each commit?